### PR TITLE
Prepare core impl for further extension

### PR DESCRIPTION
**Pull request by A248**

### CODE REVIEW

**Review by auriium — COMMENTED** [review]
looks good

File: core-implementation/src/main/java/gg/solarmc/loader/impl/LoginHandler.java [hunk 1]
```diff
@@ -43,46 +41,30 @@
...
-				for (DataGroup<?, ?> group : groups) {
-					storedData.put(group.key(), function.loadData(group.loader(), transaction, userId));
-				}
-				// It is okay that the transaction is not committed, because data loading is read-only
-			} catch (SQLException ex) {
-				throw new UncheckedSQLException("Unable to log in user", ex);
-			}
-			return new SolarPlayerImpl(storedData, userId, mcUuid);
-		});
+	public SolarPlayer loginUserNow(Transaction transaction, int userId, UUID mcUuid) {
```
Comment (Auriium) [inline core-implementation/src/main/java/gg/solarmc/loader/impl/LoginHandler.java:62]: I'm confused at the need for two of these

File: core/src/main/java/gg/solarmc/loader/data/DataLoader.java [hunk 2]
```diff
@@ -24,21 +24,23 @@
 public interface DataLoader<D> {
 
 	/**
-	 * Performs the initial data loader for a new user.
+	 * Performs the data load for a specific user. Implementations should accomodate
```
Comment (auriium) [inline core/src/main/java/gg/solarmc/loader/data/DataLoader.java:27]: This is so that we do not have to keep all dataloaders loaded on bungee, correct?

Comment (A248) [reply core/src/main/java/gg/solarmc/loader/data/DataLoader.java:27]: On velocity, correct

**Pull request merged**